pkg/controller/cassandrarestore/status.go

```diff
@@ -0,0 +1,54 @@
+package cassandrarestore
+
+import (
+	"context"
+	"fmt"
+
+	api "github.com/Orange-OpenSource/casskop/pkg/apis/db/v1alpha1"
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func UpdateRestoreStatus(c client.Client, restore *api.CassandraRestore, status api.CassandraRestoreStatus, reqLogger logr.Logger) error {
+	typeMeta := restore.TypeMeta
+
+	restore.Status = status
+
+	err := updateRestoreStatus(c, restore)
+
+	if err != nil {
+		if !apierrors.IsConflict(err) {
+			return fmt.Errorf("could not update CR state")
+		}
+
+		err := c.Get(context.TODO(), types.NamespacedName{
+			Name: restore.Name,
+			Namespace: restore.Namespace,
+		}, restore)
+
+		if err != nil {
+			return fmt.Errorf("could not get config for updating status")
+		}
+		restore.Status = status
+
+		err = updateRestoreStatus(c, restore)
+
+		if err != nil {
+			return fmt.Errorf("could not update Restore state")
+		}
+	}
+	// update loses the typeMeta of the config that's used later when setting ownerrefs
+	restore.TypeMeta = typeMeta
```

Comment:
Is it something I need to do also ?
Do we need to do it everywhere now ?

Reply:
Not sure ... Need to de some test at integration phase to be sure ?